src/metax_api/services/catalog_record_service.py

```diff
@@ -526,27 +528,115 @@ def validate_reference_data(cls, research_dataset, cache):
             raise ValidationError(errors)
 
     @classmethod
-    def strip_catalog_record(cls, catalog_record):
+    def remove_contact_info_metadata(cls, rd):
         """
-        Strip the catalog record of any confidential/private information not supposed to be
-        available for the general public.
+        Strip research dataset of any confidential/private contact information not supposed to be available for the
+        general public
 
-        Accepts as paramater either a single cr object (dict), or a list of objects.
+        :param rd:
+        :return: research dataset with removed contact information
+        """
+        return remove_keys_recursively(rd, ['email', 'telephone', 'phone'])
+
+    @classmethod
+    def check_and_remove_metadata_based_on_access_type(cls, rd):
+        """
+        If necessary, strip research dataset of any confidential/private information not supposed to be
+        available for the general public based on the research dataset's access type.
+
+        :param rd:
+        :return: research dataset with removed metadata based on access type
+        """
 
-        :param catalog_record:
+        access_type_id = cls.get_research_dataset_access_type(rd)
+        if access_type_id == ACCESS_TYPES['open']:
+            pass
+        elif access_type_id == ACCESS_TYPES['embargoed']:
+            try:
+                access_rights_available = cls.get_research_dataset_embargo_available(rd)
+                embargo_time_passed = now_is_later_than_datetime_str(access_rights_available)
+            except Exception as e:
```

Comment:
The timestamp in access_rights.available is validated during research_dataset schema validation, so in theory it should never crash?

In the unlikely case that it comes to that, serving the filtered dataset is better than crashing, but an exception should be logged at least (crashes due to bugs), rather than warning. Not that we have any log file analytics in place yet anyway to notice it... Would be a rather severe bug still, since the dataset is never going to be out of embargo, unless someone figures out to manually fix the date string.